src/PandoraMonitoring.cc

```diff
@@ -991,6 +996,49 @@ void PandoraMonitoring::ViewEvent()
 
 //------------------------------------------------------------------------------------------------------------------------------------------
 
+void PandoraMonitoring::SaveAndViewEvent(const std::string savePath)
+{
+    // TODO: We should be checking the path exists/check SavePic is capable of
+    // dealing with the case it isn't.
+
+    this->InitializeEve();
+    // TODO: For whatever reason, the first save has a different camera angle in the 3D view.
+    // This is then sorted for the rest of the events.
+    m_pEveManager->FullRedraw3D(kTRUE, kTRUE);
+
+    // Force a redraw of the event, to avoid any issues where Eve has not updated in time.
+    gSystem->ProcessEvents();
+
+    int count = 0;
+
+    // Pull out each display type and save an image of it.
+    for (auto viewer : m_pEveManager->GetViewers()->RefChildren())
+    {
+        auto eveViewer = dynamic_cast<TEveViewer*>(viewer);
+        eveViewer->GetGLViewer()->SavePictureUsingFBO(
+                savePath + "/event_" + std::to_string(m_eventDisplayCounter) +
+                "_" + DISPLAY_NAMES[count] + ".png",
```

Comment:
The documentation for `SavePictureUsingFBO` ([here](https://root.cern.ch/doc/v614/classTGLViewer.html#a845ce2f68d18b33968b0a10bafcc010c)) says `gif, gif+, jpg, png` are valid extensions here, of which I think `png` is the most useful.

Potentially could let the extension be user configurable, but not sure if its needed. For context, an event display currently as `png` is about 10kB.

Reply:
Just a thought, but it might be nice to have a configurable option for the output image file type (default being png is probably sensible though)?

Reply:
I'd hoped we'd be able to support png and pdf here, as (arguably) the two most useful formats. It seems that the pdf mechanism is different, but it'd be interesting to see if it's easy to achieve as this allows for a vector output. Otherwise, I'd agree that we'd likely only want/need png, but could still consider a configurable option. This might be by exposing that choice to the calling algorithm, e.g. by asking for an output file name with extension. Thanks!